File: app/src/main/java/com/shevelev/comics_viewer/dal/IOptionsDal.java
package com.shevelev.comics_viewer.dal;

import com.shevelev.comics_viewer.dal.dto.Option;

import java.util.List;

/**
 * Dal for innerOptionsCollection
 */
public interface IOptionsDal
{
    /**
     * @return false - fail
     */
    boolean update(List<Option> optionsToAdd, List<Option> optionsToUpdate);

    /**
     * @return false - fail
     */
    boolean delete(int[] keys);

    /**
     * @return null - fail
     */
    List<Option> getAll();
}

File: app/src/main/java/com/shevelev/comics_viewer/dal/IOptionsDal.kt
package com.shevelev.comics_viewer.dal

import com.shevelev.comics_viewer.dal.dto.Option

/**
 * Dal for innerOptionsCollection
 */
interface IOptionsDal {
    /**
     * @return false - fail
     */
    fun update(optionsToAdd: List<Option?>?, optionsToUpdate: List<Option?>?): Boolean

    /**
     * @return false - fail
     */
    fun delete(keys: IntArray?): Boolean

    /**
     * @return null - fail
     */
    val all: List<Option?>?
}